src/lib.rs

```diff
@@ -2340,7 +2340,40 @@ impl Connection {
             }
         }
 
-        // Create PING for PTO probe.
+        // Try to retransmit old frames on PTO, if there is space left.
+        if self.recovery.loss_probes[epoch] > 0 && left >= 1 && !is_closing {
+            let unacked_iter = self.recovery.sent[epoch]
+                .iter_mut()
+                // Skip packets that have already been acked or lost and packets
+                // that are not in-flight.
+                .filter(|p| p.in_flight && p.time_acked.is_none() && p.time_lost.is_none());
+
+            'unacked_iter: for unacked in unacked_iter {
+                for frame in &unacked.frames {
+                    // Only retransmit CRYPTO and STREAM frames.
+                    match frame {
+                        frame::Frame::Crypto { .. } |
+                        frame::Frame::Stream { .. } => (),
+
+                        _ => continue,
+                    }
+
+                    // Skip frame if it's too big.
+                    if left < frame.wire_len() {
+                        break 'unacked_iter;
+                    }
+
+                    push_frame_to_pkt!(frames, frame.clone(), payload_len, left);
```

Comment:
I think we need to check the return value of push_frame_to_pkt!()

Reply:
We already check that the frame can fit in the packet above this, so we don't need to check for it again.